### PR TITLE
Docker entrypoint to work with commands and arguments

### PR DESCRIPTION
**Pull request by vvaltia**
- Commands with arguments may now be used with docker image
 - To ensure backwards compability, commands with arguments
   may be included in COMMANDS environment variable using
   parenthis (). e.g.
   COMMANDS="migrate (api-key add <api-key>) run"

Signed-off-by: Valtteri Valtia <valtteri.valtia@csc.fi>

# Definition of Done / Review checklist

## Reviewability
- [ ] link to issue
- [ ] note if PR is on top of other PR
- [ ] note if related change in rems-deploy repo
- [ ] consider adding screenshots for ease of review

## API
- [ ] API is documented and shows up in Swagger UI
- [ ] API is backwards compatible or completely new
- [ ] Events are backwards compatible

## Documentation
- [ ] update changelog if necessary
- [ ] add or update docstrings for namespaces and functions
- [ ] components are added to guide page
- [ ] documentation _at least_ for config options (i.e. docs folder)
- [ ] ADR for major architectural decisions or experiments

## Different installations
- [ ] new configuration options added to rems-deploy repository
- [ ] instance specific translations (i.e. LBR kielivara)

## Testing
- [ ] complex logic is unit tested
- [ ] valuable features are integration / browser / acceptance tested automatically

## Accessibility
- [ ] all icons have the aria-label attribute
- [ ] all fields have a label
- [ ] errors are linked to fields with aria-describedby
- [ ] contrast is checked
- [ ] conscious decision about where to move focus after an action

## Follow-up
- [ ] new tasks are created for pending or remaining tasks
- [ ] no critical TODOs left to implement

### CODE REVIEW

File: docker-entrypoint.sh [hunk 1]
```diff
@@ -15,18 +19,38 @@ if [ ! -z ${certfile} ] && [ "${certfile}" != "null" ] ; then
...
+        if [[ "${comarg}" == *\) ]] ; then
+        parameters=false
+        else
+            continue
+        fi
+    # Handle run command
+    elif [ "${comarg}" = "run" ] ; then
+        cmd_prefix="exec"
+        cmd=""
+    # All other commands
```
Comment (opqdonut) [inline docker-entrypoint.sh:41]: I commend your effort of writing a parser in bash but I'd rather not have one :sweat_smile: 

How about introducing a new environment variable (for example `CMDS`) that would contain a ;-separated list of commands with arguments?

Comment (vvaltia) [reply docker-entrypoint.sh:41]: Sure. I thought about that, but I came to a conclusion that being able to use the same parameter would be simpler from the user's perspective. Ill try to come up with a more descriptive name than `CMDS` though.

Comment (opqdonut) [reply docker-entrypoint.sh:41]: I don't think our docker images have that many users yet, so we might even be able to drop the old `COMMANDS` after a release or two

File: docker-entrypoint.sh [hunk 2]
```diff
@@ -15,18 +18,25 @@ if [ ! -z ${certfile} ] && [ "${certfile}" != "null" ] ; then
             -new $(head /dev/urandom | tr -dc A-Za-z0-9 | head -c 20)
 fi
 
-for COMMAND in $COMMANDS
+if [ "${CMD}" ] ; then
+  IFS=';' read -r -a cmd_array <<< "${CMD}"
```
Comment (opqdonut) [inline docker-entrypoint.sh:22]: :ok_hand:

File: docs/installing-upgrading.md [hunk 3]
```diff
@@ -65,6 +65,12 @@ Some pointers for common tasks:
     docker-compose -f docker-compose-build.yml run --rm -e COMMANDS="migrate test-data" app
     docker-compose -f docker-compose-build.yml up -d app
 
+### COMMANDS environment variable may be populated with any rems commands
+
+- Empty `COMMANDS` variable as well as `run` command starts rems
+- For commands with arguments, use parenthis (). Examples:
+    COMMANDS="migrate (api-key add <api-key>)"
+    COMMANDS="migrate test-data (grant-role <role> <userid>) run"
```
Comment (opqdonut) [inline docs/installing-upgrading.md:73]: this still refers to the old syntax

**Review by opqdonut — CHANGES_REQUESTED** [review]

File: docs/installing-upgrading.md [hunk 4]
```diff
@@ -48,23 +48,31 @@ Some pointers for common tasks:
...
     lein uberjar
     docker-compose -f docker-compose-build.yml build
     docker-compose -f docker-compose-build.yml up -d db
-    docker-compose -f docker-compose-build.yml run --rm -e COMMANDS="migrate test-data" app
+    docker-compose -f docker-compose-build.yml run --rm -e CMD="migrate;test-data" app
     docker-compose -f docker-compose-build.yml up -d app
 
+### CMD environment variable may be populated with any rems commands
+
+Command `run` as well as empty `CMD` (and `COMMANDS`) parameter starts rems. Example `CMD` varibale values:
```
Comment (opqdonut) [inline docs/installing-upgrading.md:70]: "varibale"

I think the structure of the sentence could be a bit clearer, perhaps something like:

> You can use the CMD environment variable to specify a sequence of REMS administration commands to run. An empty CMD or the command `run` starts the REMS server.

Comment (vvaltia) [reply docs/installing-upgrading.md]: Updated

**Review by opqdonut — APPROVED** [review]

**Comment by opqdonut** [comment]
oh one more thing, please add a line in CHANGELOG.md as well!

**Review by opqdonut — APPROVED** [review]

**Pull request merged**